server/src/main/java/org/candlepin/controller/SuspendModeTransitioner.java

```diff
@@ -0,0 +1,232 @@
+     * When failedAttempts = 1, then the delay will be 15
+     * When failedAttempts = 2, then the delay will be 45
+     * etc.
+     *
+     */
+    private int delayGrowth;
+    /**
+     * Delay in seconds. SuspendModeTransitioner will wait this amount
+     * of seconds between periodic checks for Qpid connectivity.
+     */
+    private int initialDelay;
+    /**
+     * Maximum delay that this transitioner can wait between executions
+     */
+    private int maxDelay = 0;
+    private int failedAttempts = 0;
+    /**
+     * Single threaded periodic task.
+     */
+    private ScheduledExecutorService execService;
+    private ModeManager modeManager;
+    private QpidQmf qmf;
+    private QpidConnection qpidConnection;
+
+    @Inject
+    public SuspendModeTransitioner(Configuration config, ScheduledExecutorService execService) {
+        this.execService = execService;
+        delayGrowth = config.getInt(ConfigProperties.QPID_MODE_TANSITIONER_DELAY_GROWTH);
+        initialDelay = config.getInt(ConfigProperties.QPID_MODE_TRANSITIONER_INITIAL_DELAY);
+        maxDelay = config.getInt(ConfigProperties.QPID_MODE_TRANSITIONER_MAX_DELAY);
+    }
+
+    /**
+     * Other dependencies are injected using method injection so
+     * that Guice can handle circular dependency between SuspendModeTransitioner
+     * and the QpidConnection
+     */
+    @Inject
+    public void setModeManager(ModeManager modeManager) {
+        this.modeManager = modeManager;
+    }
+
+    @Inject
+    public void setQmf(QpidQmf qmf) {
+        this.qmf = qmf;
+    }
+
+    @Inject
+    public void setQpidConnection(QpidConnection qpidConnection) {
+        this.qpidConnection = qpidConnection;
+    }
+
+    /**
+     * Enables to run the transitioning logic periodically.
+     */
+    public void startPeriodicExecutions() {
+        log.info("Starting Periodic Suspend Mode Transitioner " +
+            "with delay grow factor of {}s and initial delay {}s ",
+            delayGrowth, initialDelay);
+        schedule();
+    }
+
+    /**
+     * Schedules next execution of the Suspend Mode check. The delay, in seconds,
+     * grows as the failedAttempts grow. The formula is:
+     *
+     *  delay = initialDelay * (delayGrowth ^ failedAttempts)
+     *
+     *  Maximal delay is configurable, -1 means there is no bound
+     *
+     *  When delayGrowth is set to 1, the delay wont grow.
+     */
+    private void schedule() {
+        int delay = initialDelay * (int) Math.pow(delayGrowth, failedAttempts);
+
+        if (maxDelay != -1 && delay > maxDelay) {
+            log.debug("Maximum delay {} reached", maxDelay);
+            delay = maxDelay;
+        }
+
+        log.debug("Next Transitioner check will run after {} seconds", delay);
+        if (failedAttempts > 3) {
+            log.info("SuspendModeTransitioner failed to reconnect to the Qpid Broker " +
+                "{} times, backing off with next reconnect {} seconds", failedAttempts,
```

Comment:
I kind of like it. Its easy to then search in the logs for e.g. the first time the transitioner failed to reconnect etc.